src/sentry/eventstream/kafka/backend.py

```diff
@@ -258,88 +300,89 @@ def handle_shutdown_request(signum: int, frame: Any) -> None:
         signal.signal(signal.SIGINT, handle_shutdown_request)
         signal.signal(signal.SIGTERM, handle_shutdown_request)
 
-        i = 0
-        while not shutdown_requested:
-            message = consumer.poll(0.1)
-            if message is None:
-                continue
+        def run_processing_loop() -> None:
+            """
+            Runs the processing loop of the post process forwarder. The loop runs until one of the following events
+            happens:
+            1. Commit batch size is reached
+            2. Commit batch timeout is reached.
+            When either of those events occur, we wait for all pending work to complete and then commit the offsets
+            to kafka.
+            """
+            nonlocal shutdown_requested
+            i = 0
+            while not shutdown_requested:
+                if self._pending_futures and (
+                    (i % commit_batch_size == 0)
```

Comment:
Why mod? Seems to be a bit different to what line 307 suggests

Reply:
This was just brought over by the refactor. I changed it to check for equality with commit_batch_size now